pages/concepts-limit-order-book-and-matching.md

```diff
@@ -0,0 +1,86 @@
+# Limit Order Book and Matching
+
+This document outlines the key differences between centralized exchanges and dYdX Chain, focusing on the decentralized limit order book and matching engine.
+
+
+## Blockchain Overview
+<!-- TODO: Move this to a general explainer section and link to it -->
+
+dYdX Chain is a p2p blockchain network using [CosmosSDK](https://github.com/cosmos/cosmos-sdk) and [CometBFT](https://github.com/cometbft/cometbft) (formerly Tendermint).
+
+Anyone can run a full node using the open-source software. Full nodes with sufficient delegated governance tokens participate in block building as validators.
+
+The software repository is: https://github.com/dydxprotocol/v4-chain/
+
+## Limit Order Book
+Each full node in the network maintains an in-memory order book, which undergoes state changes in real time as traders submit order instructions.
+
+Block proposers use trades from their local order book to build blocks, with matches generated by price-time priority. Since message arrival order varies between nodes, the order book may differ across the network at any given point in time. To address this, nodes partially sync their local books with each new consensus-committed block.
+
+Clients can subscribe to a node's book state using the [Full Node Streaming API](./api_integration-full-node-streaming.md).
+
+## Matching and Block Processing
+
+The order matching logic is broadly similar to centralized exchanges, with some key differences:
+
+1. On receiving a cancel instruction:
+    - The node cancels the order unless it's already matched locally.
+    - The cancel instruction is stored until it expires (based on the [GTB field](https://github.com/dydxprotocol/v4-chain/blob/4780b4cba2cab75e0af5675c3e87e551162ecf33/proto/dydxprotocol/clob/tx.proto#L90)).
+
+2. On receiving an order:
+    - The order fails to place if it has already been cancelled.
+    - Otherwise, it is matched and/or placed on the order book, with optimistic matches stored locally.
+
+3. Validator nodes propose blocks that include all their local matches.
+
+4. When processing a new block:
+    - The node starts from the state of the prior block (the local state used to propose is ignored).
```

Comment:
technically it's not ignored, it's temporarily removed from the local orderbook

Reply:
hmm yeah is there a better way to explain this without introducing new concepts?

Reply:
```
4. When processing a new block:
    - The node state is reset to the state of the prior block.
    - Optimistic local orderbook changes are temporarily removed from the node's orderbook
        - This resets the orderbook to a common state (prior block state) shared between all nodes.
    - The proposed block’s changes are applied.
    - The node replays its previously removed optimistic local orderbook changes on top of current state, during which:
        - Cancels are preserved.
        - Orders matched in the prior local state are re-placed.
        - Orders may match differently, fail to place due to cancellation, or not match at all in the new state.
```

Reply:
feel free to change